Refactor HTML generation

zymeth25 opened the pull request:
@picandocodigo perhaps you can use this to fix #350 ?

I will write a few tests for this later so that it's finished.

picandocodigo commented on include/lcp-catlist.php:
```diff
@@ -297,128 +309,134 @@ public function update_page($page) {
   }
 
-  public function get_content($single){
+  public function get_content($single) {
```
I almost feel like we shouldn't even have this function, or the parameter. If people are needing to show the content, it's almost reimplementing the blog functionality... It's been there for too long to remove it now, but it's just a thought, I don't like having this feature :stuck_out_tongue_closed_eyes:

zymeth25 replied:
I agree, WP is not intended to be used this way, a plugin should not run another main query to create a blog page. But I remember looking through support topics 2 years ago, many people use this 😆 I guess it's because WP expects you to customize blog page by editing the theme, but if you can't edit PHP you can use this plugin 😉 It's also a pain to maintain because output can differ based on theme and WP version.

zymeth25 commented on tests/catlistdisplayer/test-getPostTitle.php:
```diff
@@ -0,0 +1,277 @@
+    );
+  }
+
+  public function test_no_link_title_with_class() {
+
+    $displayer = new CatListDisplayer(
+      array_merge(self::$atts,
+        ['link_titles' => 'false', 'title_class' => 'my-class']
+      )
+    );
+
+    /*
+    How should the plugin behave when `link_titles=no` and `title_class=something` ?
+    Should it just ignore the class or should it add a default tag. If a default tag,
+    should it be <span>?
+     */
+  }
```
@picandocodigo As you can see I've started fixing #350 by writing a test suite that `get_post_title` should pass but need to make a decision about this particular test case before writing assertions. What do you think?

picandocodigo replied:
@zymeth25 I think using `span` makes sense :+1:

zymeth25 replied:
Yeah, sounds good.

zymeth25 commented:
Finished!

Fixed #350 

Most of the changes in `lcp-catlist.php` are just whitespace changes so you can filter them out.

Refactored HTML generation with the new wrapper method `to_html()`.

Added tests for `get_post_title()` that cover all features and shortcode parameters necessary. Tests are not very in-depth but are enough to make sure everything works.
